static/src/javascripts/projects/commercial/modules/braze.js

```diff
@@ -0,0 +1,62 @@
+// @flow
+
+import { onIabConsentNotification } from '@guardian/consent-management-platform';
+import config from 'lib/config';
+import { getUserFromApi } from '../../common/modules/identity/api';
+
+
+const getBrazeUuid = (): Promise<string> =>
+    new Promise(resolve => {
+        getUserFromApi(user => {
+            if (user && user.privateFields && user.privateFields.brazeUuid){
+                resolve(user.privateFields.brazeUuid)
+            }
+        })
+    })
+
+const hasRequiredConsents = (): Promise<void> =>
+    new Promise(resolve => {
+        onIabConsentNotification(state => {
+            if (state[1] && state[2] && state[3] && state[4] && state[5]) {
+                resolve()
```

Comment:
I guess you want to reject this promise otherwise it might just never resolve?

Reply:
A quick note to mention that this does not currently support CCPA or the (upcoming) TCFv2 framework. Please let us know if you require help to set this up. Some PR that might be of interest below:
- #22819 
- #22818

Reply:
@gtrufitt good spot! It was actually intentional that we're not rejecting here (my thinking was that it's not problematic/exceptional if the consents aren't given, but we can't proceed any further if that's the case, hence not resolving). I'd hadn't occurred to me that having an un-(resolved|rejected) promise hanging around might be problematic, I'll have a read of that thread! Having thought about it though, I can see this could be a bit weird/confusing for someone reading this code in future since it's definitely a bit unconventional, so rejecting if the consents aren't given probably a better approach all round.

Reply:
@mxdvl Ah yep it's definitely on my radar and this PR won't be merged until we support CCPA/TCFv2. Thanks for the PR links, they look useful. Having taken a quick look it seems like listening on consent changes is the same but we can detect CCPA from `state.ccpa` in the callback?

Reply:
@tjmw that's mostly right, but as of next week there will only be the new CCPA/TCFv2 interface.

For CCPA, the in the callback on `onConsentChange`, you need to check `state.ccpa` exists/is set. Then you can confirm `state.ccpa.doNotSell === false` and resolve.

Reply:
We are planning to officially release the CMP v4 next Wednesday—12 August—and will be on call to assist if need be!